lib/Raven/Client.php

```diff
@@ -112,13 +113,26 @@ class Client
     protected $config;
 
     /**
+     * @var HttpAsyncClient The HTTP client
+     */
+    private $httpClient;
+
+    /**
+     * @var MessageFactory The message factory
+     */
+    private $messageFactory;
+
+    /**
      * Constructor.
      *
-     * @param Configuration $config The client configuration
+     * @param Configuration   $config     The client configuration
+     * @param HttpAsyncClient $httpClient The HTTP client
      */
-    public function __construct(Configuration $config)
+    public function __construct(Configuration $config, HttpAsyncClient $httpClient, MessageFactory $messageFactory)
```

Comment:
No-no, wait. **Typehints** have nothing to do with this question. Of course typehints are need. More than that, hints in your IDE are need. In my projects I even need to create separate file with objects description [[1]](https://github.com/nokitakaze/php-mutex/blob/master/src/phpdoc.php) [[2]](https://github.com/nokitakaze/php-keyvalue/blob/master/src/phpdoc.php) so my Idea shows me properties of "stdClass".

I say you could use all params as separate properties **in** `Configuration`